src/Nerdbank.Streams/MultiplexingStream.Formatters.cs

```diff
@@ -410,6 +410,51 @@ internal override async Task<(FrameHeader Header, ReadOnlySequence<byte> Payload
                 }
             }
 
+            /// <summary>
+            /// Creates a payload to sent alongside <see cref="ControlCode.ChannelTerminated"/>.
+            /// </summary>
+            /// <param name="exception">The exception to send to the remote side if there is one.</param>
+            /// <returns>The payload to send when a channel gets terminated.</returns>
+            internal ReadOnlySequence<byte> SerializeException(Exception? exception)
+            {
+                var sequence = new Sequence<byte>();
+                var writer = new MessagePackWriter(sequence);
+
+                // Only write the exception if there is an exception and the user requested to write it
+                if (exception != null)
+                {
+                    writer.WriteArrayHeader(1);
+                    writer.Write(exception.Message);
+                }
+                else
+                {
+                    // Since there is no error send an empty payload
+                    writer.WriteArrayHeader(0);
+                }
+
+                writer.Flush();
+                return sequence;
+            }
+
+            /// <summary>
+            /// Gets the error message in the payload if there is one.
+            /// </summary>
+            /// <param name="payload">The payload that could contain an error message.</param>
+            /// <returns>The error message in this payload if there is one, null otherwise.</returns>
+            internal string? DeserializeException(ReadOnlySequence<byte> payload)
```

Comment:
The main thing was that I was not sure that the exception type would be as well as what the error message would be like. Since I already do this in the method that calls Deserialize, I will move those changes into this method.